src/stores/useSocialMediaStore/useSocialMediaStore.types.ts

```diff
@@ -6,14 +6,21 @@ import { Tabs } from '~components/Tabber/Tabber.types';
 
 export type StoreAccount = Account & { valid: boolean };
 
+export type SocialMediaAccounts = {
+  account: Account[];
+  socialMediaId: Account['socialMediaId'];
+}[];
+
+export type NewAccount = Omit<StoreAccount, 'id'>;
+
 export type SocialMediaState = {
   accounts: {
-    data: StoreAccount[] | null;
+    data: StoreAccount[];
     error: string;
     loading: boolean;
   };
 
-  addAccount: () => Promise<void>;
+  addSocialMedia: (newAccount: NewAccount) => Promise<GenericObject>;
```

Comment:
Alterar também aqui para o novo type
```suggestion
  addSocialMedia: (newAccount: NewAccount) => Promise<NewAccount>;
```